metricbeat/module/etcd/leader/leader.go

```diff
@@ -71,14 +80,46 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 // format. It publishes the event which is then forwarded to the output. In case
 // of an error set the Error field of mb.Event or simply call report.Error().
 func (m *MetricSet) Fetch(reporter mb.ReporterV2) error {
-	content, err := m.http.FetchContent()
+	res, err := m.http.FetchResponse()
+	if err != nil {
+		return errors.Wrap(err, "error fetching response")
+	}
+	defer res.Body.Close()
+
+	content, err := ioutil.ReadAll(res.Body)
 	if err != nil {
-		return errors.Wrap(err, "error in http fetch")
+		return errors.Wrapf(err, "error reading body response")
+	}
+
+	if res.StatusCode == http.StatusOK {
+		reporter.Event(mb.Event{
+			MetricSetFields: eventMapping(content),
+			ModuleFields:    common.MapStr{"api_version": apiVersion},
+		})
+		return nil
+	}
+
+	// Errors might be reported as {"message":"<error message>"}
+	// let's look for that structure
+	var jsonResponse map[string]interface{}
+	if err = json.Unmarshal(content, &jsonResponse); err == nil {
+		if retMessage := jsonResponse[msgElement]; retMessage != "" {
+			// there is an error message element, let's use it
+
+			// If a 403 is returned and {"message":"not current leader"}
+			// do not consider this an error
+			// do not report events since this is not a leader
+			if res.StatusCode == http.StatusForbidden &&
+				retMessage == msgValueNonLeader {
+				return nil
```

Comment:
maybe add a debug message?

Reply:
Agree on a debug/info level log here 👍

Reply:
Cool, thanks for the review.

I'm ok with all debug info, but let me clarify. An etcd cluster is usually 3 or 5 instances, only one of those instances will show an event, not the other 2, we can consider that _if_ statement above the _normal_ path.
Not an error, not an unexpected path.

I'll add a **Debug** log in there if that's ok with you.

Reply:
I agree it's not an error, still some users may not understand why they are not getting metrics from a node where this module is running :+1:, debug log will help them understand that